### DISCUSSION TITLE
Rethink order management - make order statuses more clear and add more control

### DISCUSSION THREAD

**Discussion by madsoliver** [PrestaShop/PrestaShop#31038] [Ideas]
### The problem

**We get a lot of questions like:** 
- What does this order state do?
- Why is stock not reduced?
- Why cant I get an invoice?
- Why cant I get a delivery slip?
- Why cant I change the order products after I printed delivery slip?
- Why is the payment still just reserved and not captured?
- Why/when is the payment refunded or cancelled?
- When is the order sent to accounting?
- Why can I still edit the order, when it is sent to accounting?
- Why cant I longer edit the order - the customer wants?

And the questions are always hard to answer - the entire order states system in PS, along with all checkboxes inside the Order State options page, seems a bit random. No offence intended. I just think this is something that was thought through from the beginning, but as PS has evolved (and so has internet), the entire order status / order management system needs to be thought trough again. 

### The solution

Rethinking this would be a good idea. 

I already made an issue: #19043

**But also adding this:**

There should also be a radio-select for each status:  `reserve ordered products` | `subtract ordered products from physical quantity` | `cancel/return previous subtracted/reserved stock` | `do nothing with product stock`.

**Also I would love checkboxes for:**
`Capture payment for entire order` - Could be paired up with payment modules.
`Cancel payment for entire order` - Could be paired up with payment modules.
`Refund payment for entire order` - Could be paired up with payment modules.
`Send to accounting` - a "status" that would be paired up with accounting-integration-modules.

**All checkboxes should trigger hooks like:**
`hookCapturePayment`, `hookCancelPayment` and so on.
Most payment/accounting-modules uses `actionOrderStatusPostUpdate` with if-statements to look for the correct `id_order_state`, but if we add for example `hookCapturePayment`, most modules could be set up much easier. Off course all modules can choose if they want to implement the new hooks, or they can use the allready existing `actionOrderStatusPostUpdate`. 

### Alternatives

Maybe if the above is implemented, it maybe should be considered a "experimental feature", so that merchants can choose when to impelent it. They would need to go throught their order states and make choices after activating the "New order state system"

### Additional context

**Very related to:**
#17598

### Do you plan to work on this subject?

- [X] I'm willing to contribute a [formal specification](https://github.com/PrestaShop/prestashop-specs).
- [X] I'm willing to provide any wireframes or design assets required for this feature.
- [X] I'm willing to submit a Pull Request that implements this feature.
- [X] I'm willing to help verify that the implemented feature works as intended and produces no unintended side effects.

**Comment by matks** [comment]
This can help https://devdocs.prestashop-project.org/8/development/orders-lifecycle/

This page has received great work from @thomasnares

**Reply by madsoliver** [reply]
I read the entire article now and it is a good piece of documentation 🙂 
However, I was thinking more of what happens **after** the process in the article:

- What causes stock to be reserved
- What subtracts stock?
- What causes payment to be captured (for payment options that have that)
- What causes delivery slip to be created
- What causes invoice to generated?
- What locks an order from being edited (add, remove products, shipping and discounts)
- What happens if products are changed, is shipping price and options updated? Can the employee have a choice if this should happen for each order edit?
- What happens with order payment when changing the order content? Should the employee add transactions (+ or -)? 
- Can an Invoice and delivery slip be "trusted"  if you print them, or can they be unaligned with the order (or even changed)?
- What causes an order to be refunded (for payment providers that have that)
- What causes stock reservation to be removed?
- What causes stock to be added back?

**Comment by Hlavtox** [comment]
@madsoliver I had a very simmilar idea - https://github.com/PrestaShop/PrestaShop/issues/24084, do you have same things in mind?
